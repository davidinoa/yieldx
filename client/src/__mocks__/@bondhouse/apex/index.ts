import { AccountStatus } from "@bondhouse/apex";
import { MOCK_APEX_ACCOUNT } from "./__fixtures__";
// eslint-disable-next-line @typescript-eslint/no-empty-function,no-shadow
function ApexAccountsApi() {}

ApexAccountsApi.prototype.getApexAccountByInvestorId = jest.fn(() =>
  Promise.resolve({
    data: {
      apexAccount: MOCK_APEX_ACCOUNT
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function,no-shadow
function StatementsApi() {}

StatementsApi.prototype.getAccountStatements = jest.fn(() =>
  Promise.resolve([
    {
      apexAccountId: "28b5b427-053b-4ad8-9acb-05a5e7bbe614",
      statmentDate: "12/31/2014",
      documentType: "Account Statement",
      statementURL:
        "https://UATEDocuments.apexclearing.com/RedirectConfirmURL.aspx?ds=1DC69CF82426F431D4535F6549270DAC58EEB05D61A8209E49CAAE3BC76EE8B866CEDC5A04A68284717045A7352477CC49D5E186BA7CA707C0EF8A692C654BED5732532662D182181AAF4E3C7F5B219956BF0F1B6F57C588C0465DF32D3A5EB58EFB4CFF188B33275E8BFB631B97C80DFEB0838AECFC00DF537E96EB28868EFF10660E83DD1A6C5808E6DED036C15E25336F448E9F69DF30118E6960DD1F98D4B8E4714BCAE3D1435A79505BC3FB730B09FEA86B8E3DEB1F8DDAF65E746FBE833308839C1FA0381057F4E71EFC7E14AF5DBB327B2CA848EC4CC2D036AF66C648E78AD70688C550DEB37F942CC82AEEE4F6E1BA95C082DB8D326BC86D5B5DC113060E7449A97FD2F019A66050CA0AF83B0FA50FFCFE9F5E1664C92DA902988BDF&pid=TkVXU1RNVHx2NzEyNi01Mj9yLTUyOTMtNzc1xS1PREEyLTI4MzBGQUFZLTM5NzAxLTU0MzIxNC00MjE4NzItMjE5MjAyLTg1LTc5LTI2NDMtODktMC1eATAxNAE1UEkBMzIyNzIBTUFSSyBTTEFWT05JQQExN1EwNgFXRUEBAgECAQIBAgExNDQBMTcxMDgBMTMxODQBAgECAQIBA0101816"
    }
  ])
);

export { ApexAccountsApi, AccountStatus, StatementsApi };
