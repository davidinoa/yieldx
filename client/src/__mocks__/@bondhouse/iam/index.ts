import {
  CreateInvestorMappingRequest,
  UserType,
  UserStatus
} from "@bondhouse/iam";

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function UsersApi() {}

UsersApi.prototype.getCurrentUser = jest.fn(() =>
  Promise.resolve({
    data: {
      user: {
        id: "auth0|5ec6119474dfa00cbf1d53cb",
        first: "Institution",
        last: "New User",
        userType: "ADMIN",
        email: "test4@bondhouse.com",
        terms: true,
        institutionId: "bf8a5597-9569-4510-babd-a6d9655e9f0d",
        createdAt: "2020-05-21T05:28:52.982380Z",
        updatedAt: "2020-05-21T05:28:52.982380Z",
        status: "ACTIVE",
        policies: []
      }
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function PasswordResetTicketsApi() {
  return {};
}

PasswordResetTicketsApi.prototype.createPasswordResetTicket = jest.fn(() =>
  Promise.resolve({
    data: {
      success: true
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function InstitutionInventoryMappingApi() {}

InstitutionInventoryMappingApi.prototype.getInstitutionInventoryMapping = jest.fn(
  () =>
    Promise.resolve({
      data: {
        inventoryMapping: {
          institutionId: "b17c644a-30e8-47fb-a26b-f6743786c3ba",
          inpaas: "inpaas",
          bestFit: "bestfit",
          assetExplorer: "assetexplorer"
        }
      }
    })
);
InstitutionInventoryMappingApi.prototype.createOrUpdateInstitutionInventoryMapping = jest.fn();

// eslint-disable-next-line @typescript-eslint/no-empty-function
export function InstitutionInvestorMappingApi() {}

InstitutionInvestorMappingApi.prototype.createInstitutionInvestorMapping = jest.fn(
  (
    institutionId: string,
    createInvestorMappingRequest?: CreateInvestorMappingRequest
  ) =>
    Promise.resolve({
      data: {
        investorMapping: {
          institutionId,
          investorId: createInvestorMappingRequest?.investorId || ""
        }
      }
    })
);

export { UserType, UserStatus };
