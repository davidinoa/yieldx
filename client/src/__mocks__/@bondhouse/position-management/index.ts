import {
  CreateFundTransferResponse,
  CreateOrReplaceExternalPortfolioResponse,
  LedgerEntryStatusEnum,
  PortfolioStatus,
  AppType
} from "@bondhouse/position-management";

import samplePortfolio from "util/samplePortfolio";
import usdCash from "util/usdCash";

export function PortfoliosApi() {
  return {
    async internalPortfolioFundTransfer() {
      return {
        accepted: true,
        ledgerEntry: {
          id: "",
          tradeId: "",
          portfolioId: "",
          assetId: "",
          quantity: 1,
          tradePrice: 1,
          counterPartyId: "",
          relatedPositions: [],
          status: LedgerEntryStatusEnum.SETTLED
        }
      } as CreateFundTransferResponse;
    },

    updatePortfolio: jest.fn(async () => ({
      id: "mockId",
      createdAt: new Date().toISOString(),
      updateAt: new Date().toISOString(),
      status: PortfolioStatus.TERMINATED,
      currency: "USD",
      name: "Mock Portfolio",
      positions: [usdCash]
    })),

    getPortfolio: jest.fn(async () => {
      return {
        data: {
          portfolio: {
            ...samplePortfolio,
            positions: []
          }
        }
      };
    }),

    createPortfolioForInvestor: jest.fn(async () => {
      const portfolio = {
        ...samplePortfolio,
        positions: []
      };
      delete portfolio.metadata;
      return { data: { portfolio } };
    })
  };
}

export function TradesApi() {
  return {
    getTrades: jest.fn(async () => ({
      data: {
        ledgerEntries: []
      }
    }))
  };
}

export function ExternalPortfolioApi() {
  return {
    createExternalPortfolio: jest.fn(async () => {
      return {
        created: true,
        updated: false
      } as CreateOrReplaceExternalPortfolioResponse;
    }),
    getExternalPortfolio: jest.fn(async () => {
      return {
        data: {
          portfolio: { positions: [] }
        }
      };
    })
  };
}

export { AppType, PortfolioStatus, LedgerEntryStatusEnum };
