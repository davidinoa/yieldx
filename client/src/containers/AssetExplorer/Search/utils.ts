interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
