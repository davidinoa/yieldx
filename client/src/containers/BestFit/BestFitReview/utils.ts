import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
