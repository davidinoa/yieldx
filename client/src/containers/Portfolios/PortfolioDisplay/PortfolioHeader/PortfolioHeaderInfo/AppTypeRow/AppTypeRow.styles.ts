import styled from "styled-components";

export const InfoStatus = styled.div.attrs({
  className: "col-24 d-flex",
  "data-testid": "Header-Info-Status"
})`
  h5 {
    display: inline-block;
    color: white;
    text-transform: uppercase;
    letter-spacing: 0.15em;
    font-size: 0.7rem;
    > svg {
      margin: -0.25rem 0.7rem 0 0;

      > path {
        fill: white;
      }
    }
    &.status {
      border-right: solid 1px var(--border-new);
      color: white;
      &::before {
        content: "";
        display: inline-block;
        width: 0.75em;
        height: 0.75em;
        margin-right: 0.5rem;
        border-radius: 100%;
      }
    }
          &.portfolio-status-paper::before{
        background: var(--tracking);
        }
        &.portfolio-status-live::before {
        background: var(--active);
        }
        &.portfolio-status-closed::before {
        background: var(--lightest);
        }
        &.portfolio-status-pending::before {
        background: var(--caution);
        }
  }
`;

export const EditGoalsContainer = styled.div`
  button {
   background: inherit;
   font-weight: bold;
   margin-bottom: 7px;
  font-size: 12px;
  line-height: 15px;
  padding-left: 0;
  color: var(--subtext-3);
  }
  .status-separator {
  margin-left: 10px;
  margin-top: 3px;
    height: 10px;
    width: 1.5px;
    background: var(--border-new);
  }
`;
