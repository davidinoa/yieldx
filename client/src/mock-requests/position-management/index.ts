import { createGetPortfolioHandler } from "./createGetPortfolioHandler";

export default {
  defaultGetPortfolio: createGetPortfolioHandler
};
