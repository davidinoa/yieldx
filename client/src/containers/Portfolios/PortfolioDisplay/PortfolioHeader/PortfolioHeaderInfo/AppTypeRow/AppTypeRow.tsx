import React from "react";
import { Row } from "react-bootstrap";
import { AppType } from "@bondhouse/position-management";
import { EditIcon } from "../../PortfolioHeader.styles";
import { ReactComponent as PortfolioIcon } from "../../../../../../assets/svg/portfolio.svg";
import { EditGoalsContainer, InfoStatus } from "./AppTypeRow.styles";

export type PortfolioDisplayStatus = "PAPER" | "LIVE" | "CLOSED" | "PENDING";

interface Props {
  status?: PortfolioDisplayStatus;
  appType?: AppType;
  onEditGoals?: () => void;
}

export default function AppTypeRow({ status, appType, onEditGoals }: Props) {
  return (
    <Row>
      <InfoStatus>
        {status && (
          <h5
            className={`pr-2 status portfolio-status-${status.toLowerCase()}`}
          >
            {status.toUpperCase()}
          </h5>
        )}
        <h5 className={status !== undefined ? "ml-3" : undefined}>
          <PortfolioIcon width="15" height="13" />
          {`${appType ?? "CUSTOM"} PORTFOLIO`}
        </h5>
        {onEditGoals && (
          <EditGoalsContainer className="d-flex">
            <div className="status-separator" />
            <button
              className="ml-3"
              type="button"
              data-testid="edit-name-button"
              // TODO: make edit name modal open with local state instead of url hash
              onClick={onEditGoals}
            >
              <EditIcon className="mr-2" />
              Edit goals
            </button>
          </EditGoalsContainer>
        )}
      </InfoStatus>
    </Row>
  );
}
