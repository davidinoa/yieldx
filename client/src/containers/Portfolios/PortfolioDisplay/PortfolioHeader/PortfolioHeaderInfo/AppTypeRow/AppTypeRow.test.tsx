import React from "react";
import { AppType } from "@bondhouse/position-management";
import userEvent from "@testing-library/user-event";
import AppTypeRow from "./AppTypeRow";
import { render, screen } from "../../../../../../test/utils";

describe("AppTypeRow", () => {
  it("Should render custom review portfolio", () => {
    const { asFragment } = render(
      <AppTypeRow status={undefined} appType={AppType.CUSTOM} />
    );
    expect(screen.queryByText(/custom portfolio/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render inpaas review portfolio with edit button", () => {
    const mockEditGoals = jest.fn();
    const { asFragment } = render(
      <AppTypeRow
        status={undefined}
        appType={AppType.INPAAS}
        onEditGoals={mockEditGoals}
      />
    );
    expect(screen.queryByText(/inpaas portfolio/i)).toBeInTheDocument();
    expect(screen.getByRole("button")).toHaveTextContent(/edit goals/i);
    userEvent.click(screen.getByRole("button"));
    expect(mockEditGoals).toHaveBeenCalledTimes(1);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render live portfolio status", () => {
    const { asFragment } = render(
      <AppTypeRow status="LIVE" appType={AppType.CUSTOM} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render paper portfolio status", () => {
    const { asFragment } = render(
      <AppTypeRow status="PAPER" appType={AppType.CUSTOM} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render closed portfolio status", () => {
    const { asFragment } = render(
      <AppTypeRow status="CLOSED" appType={AppType.CUSTOM} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Should render pending portfolio status", () => {
    const { asFragment } = render(
      <AppTypeRow status="PENDING" appType={AppType.CUSTOM} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
