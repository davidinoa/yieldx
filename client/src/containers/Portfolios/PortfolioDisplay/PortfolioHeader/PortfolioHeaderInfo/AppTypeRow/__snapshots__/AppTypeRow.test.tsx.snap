// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTypeRow Should render closed portfolio status 1`] = `
<DocumentFragment>
  .c0 h5 {
  display: inline-block;
  color: white;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  font-size: 0.7rem;
}

.c0 h5 > svg {
  margin: -0.25rem 0.7rem 0 0;
}

.c0 h5 > svg > path {
  fill: white;
}

.c0 h5.status {
  border-right: solid 1px var(--border-new);
  color: white;
}

.c0 h5.status::before {
  content: "";
  display: inline-block;
  width: 0.75em;
  height: 0.75em;
  margin-right: 0.5rem;
  border-radius: 100%;
}

.c0 h5.portfolio-status-paper::before {
  background: var(--tracking);
}

.c0 h5.portfolio-status-live::before {
  background: var(--active);
}

.c0 h5.portfolio-status-closed::before {
  background: var(--lightest);
}

.c0 h5.portfolio-status-pending::before {
  background: var(--caution);
}

<div
    class="row"
  >
    <div
      class="c0 col-24 d-flex"
      data-testid="Header-Info-Status"
    >
      <h5
        class="pr-2 status portfolio-status-closed"
      >
        CLOSED
      </h5>
      <h5
        class="ml-3"
      >
        <svg
          height="13"
          width="15"
        >
          portfolio.svg
        </svg>
        CUSTOM PORTFOLIO
      </h5>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppTypeRow Should render custom review portfolio 1`] = `
<DocumentFragment>
  .c0 h5 {
  display: inline-block;
  color: white;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  font-size: 0.7rem;
}

.c0 h5 > svg {
  margin: -0.25rem 0.7rem 0 0;
}

.c0 h5 > svg > path {
  fill: white;
}

.c0 h5.status {
  border-right: solid 1px var(--border-new);
  color: white;
}

.c0 h5.status::before {
  content: "";
  display: inline-block;
  width: 0.75em;
  height: 0.75em;
  margin-right: 0.5rem;
  border-radius: 100%;
}

.c0 h5.portfolio-status-paper::before {
  background: var(--tracking);
}

.c0 h5.portfolio-status-live::before {
  background: var(--active);
}

.c0 h5.portfolio-status-closed::before {
  background: var(--lightest);
}

.c0 h5.portfolio-status-pending::before {
  background: var(--caution);
}

<div
    class="row"
  >
    <div
      class="c0 col-24 d-flex"
      data-testid="Header-Info-Status"
    >
      <h5>
        <svg
          height="13"
          width="15"
        >
          portfolio.svg
        </svg>
        CUSTOM PORTFOLIO
      </h5>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppTypeRow Should render inpaas review portfolio with edit button 1`] = `
<DocumentFragment>
  .c0 h5 {
  display: inline-block;
  color: white;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  font-size: 0.7rem;
}

.c0 h5 > svg {
  margin: -0.25rem 0.7rem 0 0;
}

.c0 h5 > svg > path {
  fill: white;
}

.c0 h5.status {
  border-right: solid 1px var(--border-new);
  color: white;
}

.c0 h5.status::before {
  content: "";
  display: inline-block;
  width: 0.75em;
  height: 0.75em;
  margin-right: 0.5rem;
  border-radius: 100%;
}

.c0 h5.portfolio-status-paper::before {
  background: var(--tracking);
}

.c0 h5.portfolio-status-live::before {
  background: var(--active);
}

.c0 h5.portfolio-status-closed::before {
  background: var(--lightest);
}

.c0 h5.portfolio-status-pending::before {
  background: var(--caution);
}

.c1 button {
  background: inherit;
  font-weight: bold;
  margin-bottom: 7px;
  font-size: 12px;
  line-height: 15px;
  padding-left: 0;
  color: var(--subtext-3);
}

.c1 .status-separator {
  margin-left: 10px;
  margin-top: 3px;
  height: 10px;
  width: 1.5px;
  background: var(--border-new);
}

<div
    class="row"
  >
    <div
      class="c0 col-24 d-flex"
      data-testid="Header-Info-Status"
    >
      <h5>
        <svg
          height="13"
          width="15"
        >
          portfolio.svg
        </svg>
        INPAAS PORTFOLIO
      </h5>
      <div
        class="c1 d-flex"
      >
        <div
          class="status-separator"
        />
        <button
          class="ml-3"
          data-testid="edit-name-button"
          type="button"
        >
          <svg
            class="mr-2"
          >
            edit.svg
          </svg>
          Edit goals
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppTypeRow Should render live portfolio status 1`] = `
<DocumentFragment>
  .c0 h5 {
  display: inline-block;
  color: white;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  font-size: 0.7rem;
}

.c0 h5 > svg {
  margin: -0.25rem 0.7rem 0 0;
}

.c0 h5 > svg > path {
  fill: white;
}

.c0 h5.status {
  border-right: solid 1px var(--border-new);
  color: white;
}

.c0 h5.status::before {
  content: "";
  display: inline-block;
  width: 0.75em;
  height: 0.75em;
  margin-right: 0.5rem;
  border-radius: 100%;
}

.c0 h5.portfolio-status-paper::before {
  background: var(--tracking);
}

.c0 h5.portfolio-status-live::before {
  background: var(--active);
}

.c0 h5.portfolio-status-closed::before {
  background: var(--lightest);
}

.c0 h5.portfolio-status-pending::before {
  background: var(--caution);
}

<div
    class="row"
  >
    <div
      class="c0 col-24 d-flex"
      data-testid="Header-Info-Status"
    >
      <h5
        class="pr-2 status portfolio-status-live"
      >
        LIVE
      </h5>
      <h5
        class="ml-3"
      >
        <svg
          height="13"
          width="15"
        >
          portfolio.svg
        </svg>
        CUSTOM PORTFOLIO
      </h5>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppTypeRow Should render paper portfolio status 1`] = `
<DocumentFragment>
  .c0 h5 {
  display: inline-block;
  color: white;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  font-size: 0.7rem;
}

.c0 h5 > svg {
  margin: -0.25rem 0.7rem 0 0;
}

.c0 h5 > svg > path {
  fill: white;
}

.c0 h5.status {
  border-right: solid 1px var(--border-new);
  color: white;
}

.c0 h5.status::before {
  content: "";
  display: inline-block;
  width: 0.75em;
  height: 0.75em;
  margin-right: 0.5rem;
  border-radius: 100%;
}

.c0 h5.portfolio-status-paper::before {
  background: var(--tracking);
}

.c0 h5.portfolio-status-live::before {
  background: var(--active);
}

.c0 h5.portfolio-status-closed::before {
  background: var(--lightest);
}

.c0 h5.portfolio-status-pending::before {
  background: var(--caution);
}

<div
    class="row"
  >
    <div
      class="c0 col-24 d-flex"
      data-testid="Header-Info-Status"
    >
      <h5
        class="pr-2 status portfolio-status-paper"
      >
        PAPER
      </h5>
      <h5
        class="ml-3"
      >
        <svg
          height="13"
          width="15"
        >
          portfolio.svg
        </svg>
        CUSTOM PORTFOLIO
      </h5>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AppTypeRow Should render pending portfolio status 1`] = `
<DocumentFragment>
  .c0 h5 {
  display: inline-block;
  color: white;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.15em;
  -moz-letter-spacing: 0.15em;
  -ms-letter-spacing: 0.15em;
  letter-spacing: 0.15em;
  font-size: 0.7rem;
}

.c0 h5 > svg {
  margin: -0.25rem 0.7rem 0 0;
}

.c0 h5 > svg > path {
  fill: white;
}

.c0 h5.status {
  border-right: solid 1px var(--border-new);
  color: white;
}

.c0 h5.status::before {
  content: "";
  display: inline-block;
  width: 0.75em;
  height: 0.75em;
  margin-right: 0.5rem;
  border-radius: 100%;
}

.c0 h5.portfolio-status-paper::before {
  background: var(--tracking);
}

.c0 h5.portfolio-status-live::before {
  background: var(--active);
}

.c0 h5.portfolio-status-closed::before {
  background: var(--lightest);
}

.c0 h5.portfolio-status-pending::before {
  background: var(--caution);
}

<div
    class="row"
  >
    <div
      class="c0 col-24 d-flex"
      data-testid="Header-Info-Status"
    >
      <h5
        class="pr-2 status portfolio-status-pending"
      >
        PENDING
      </h5>
      <h5
        class="ml-3"
      >
        <svg
          height="13"
          width="15"
        >
          portfolio.svg
        </svg>
        CUSTOM PORTFOLIO
      </h5>
    </div>
  </div>
</DocumentFragment>
`;
